repa/grids/psdiffusion.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2017-2019 The repa authors
+ *
+ * This file is part of Repa.
+ *
+ * Repa is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Repa is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Repa.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "psdiffusion.hpp"
+
+#include <algorithm>
+#include <set>
+
+#include "grids/util/mpi_cart.hpp"
+#include "grids/util/mpi_graph.hpp"
+#include "util/vec_arith.hpp"
+
+#ifndef NDEBUG
+#define PSDIFFUSION_DEBUG
+#endif
+
+namespace repa {
+namespace grids {
+
+static Vec3i
+fix_periodic_edge(const Vec3i &c0, const Vec3i &c2, const Vec3i &gsize)
```

Comment:
This function needs a better name. :)
"fix_periodic..." is its intention, but what does it do?

Reply:
Updated in 14f7a25